api/src/main/java/org/apache/iceberg/TrackedFile.java

```diff
@@ -0,0 +1,268 @@
+  Types.NestedField LOCATION =
+      Types.NestedField.optional(
+          100,
+          "location",
+          Types.StringType.get(),
+          "Location of the file. Optional if content_type is 5 and deletion_vector.inline_content is not null");
+  Types.NestedField FILE_FORMAT =
+      Types.NestedField.required(
+          101,
+          "file_format",
+          Types.StringType.get(),
+          "String file format name: avro, orc, parquet, or puffin");
+  Types.NestedField PARTITION_SPEC_ID =
+      Types.NestedField.required(
+          148,
+          "partition_spec_id",
+          Types.IntegerType.get(),
+          "ID of partition spec used to write manifest or data/delete files");
+  Types.NestedField SORT_ORDER_ID =
+      Types.NestedField.optional(
+          140,
+          "sort_order_id",
+          Types.IntegerType.get(),
+          "ID representing sort order for this file. Can only be set if content_type is 0");
+  Types.NestedField RECORD_COUNT =
+      Types.NestedField.required(
+          103,
+          "record_count",
+          Types.LongType.get(),
+          "Number of records in this file, or the cardinality of a deletion vector");
+  Types.NestedField FILE_SIZE_IN_BYTES =
+      Types.NestedField.optional(
+          104,
+          "file_size_in_bytes",
+          Types.LongType.get(),
+          "Total file size in bytes. Must be defined if location is defined");
+  Types.NestedField KEY_METADATA =
+      Types.NestedField.optional(
+          131,
+          "key_metadata",
+          Types.BinaryType.get(),
+          "Implementation-specific key metadata for encryption");
+  Types.NestedField SPLIT_OFFSETS =
+      Types.NestedField.optional(
+          132,
+          "split_offsets",
+          Types.ListType.ofRequired(133, Types.LongType.get()),
+          "Split offsets for the data file. Must be sorted ascending");
+  Types.NestedField EQUALITY_IDS =
+      Types.NestedField.optional(
+          135,
+          "equality_ids",
+          Types.ListType.ofRequired(136, Types.IntegerType.get()),
+          "Field ids used to determine row equality in equality delete files. Required when content=2");
+  Types.NestedField REFERENCED_FILE =
+      Types.NestedField.optional(
+          143,
+          "referenced_file",
+          Types.StringType.get(),
+          "Location of data file that a DV references if content_type is 1 or 5."
+              + " Location of affiliated data manifest if content_type is 4 or null if delete manifest is unaffiliated");
+
+  /**
+   * Returns the path of the manifest which this file is referenced in or null if it was not read
+   * from a manifest.
+   */
+  String manifestLocation();
+
+  /**
+   * Returns the tracking information for this entry.
+   *
+   * <p>Contains status, snapshot ID, sequence numbers, and first-row-id. Optional - may be null if
+   * tracking info is inherited.
+   */
+  TrackingInfo trackingInfo();
```

Comment:
I think that this also requires a field to be defined, so we have a record of the ID used for it.

Reply:
What is `content_slice`?